replica/database.hh

```diff
@@ -1713,6 +1713,26 @@ public:
     }
 };
 
+// global_column_family_ptr provides a way to easily retrieve local instance of a given column family.
+class global_column_family_ptr {
+    distributed<replica::database>& _db;
+    table_id _id;
+private:
+    replica::column_family& get() const { return _db.local().find_column_family(_id); }
+public:
+    global_column_family_ptr(distributed<replica::database>& db, sstring ks_name, sstring cf_name)
+        : _db(db)
+        , _id(_db.local().find_column_family(ks_name, cf_name).schema()->id()) {
+    }
+
+    replica::column_family* operator->() const {
+        return &get();
+    }
+    replica::column_family& operator*() const {
+        return get();
+    }
+};
+
```

Comment:
my preference is that global_column_family_ptr remains in the source file, because it's not used elsewhere and therefore doesn't have to be exposed in a header. we did tons of work around database.hh to reduce compilation time, it should include only essential stuff. even though it's a functionality built upon replica::database, it's only used within a context of sharded sstable loader service. if you really feel like moving it into a header, perhaps we should create database_utils.hh

Reply:
yes, sorry.  I had a version where sstables_metadata was defined in distributed_loader.hh, requiring global_column_family_ptr, but I reverted that so it's no longer needed. Will remove first 2 patches in the next version